internal/controller/compute/user/user.go

```diff
@@ -196,14 +266,27 @@ func (eu *externalUser) Create(ctx context.Context, mg resource.Managed) (manage
 	}
 	cr.SetConditions(xpv1.Creating())
 
-	observed, resp, err := eu.service.CreateUser(ctx, cr.Spec.ForProvider)
+	var passw string
+	if cr.HasCredentialsSecretRef() {
+		secret, err := getCredentialsSecret(ctx, eu.client, cr.Spec.ForProvider.CredentialsSecretRef)
+		if err != nil {
+			return managed.ExternalCreation{}, errors.Wrap(err, errUserCreate)
+		}
+		passw = string(secret.Data[cr.Spec.ForProvider.CredentialsSecretRef.Key])
+		cr.Status.AtProvider.CredentialsVersion = secret.GetResourceVersion()
+	}
+
+	observed, resp, err := eu.service.CreateUser(ctx, cr.Spec.ForProvider, passw)
```

Comment:
Since this is where the condition "Is there a CredentialsSecretRef" is evaluated and acted upon, this is where the other source for the password should also be evaluated.
```suggestion
	if cr.HasCredentialsSecretRef() {
	  ...
	}
        // Deprecated: this functionality is deprecated as of v1.1.10
	else if pw := cr.Spec.ForProvider.Password; pw != ""  { 
		passw = pw
	}
	observed, resp, err := eu.service.CreateUser(ctx, cr.Spec.ForProvider, passw)
```

Reply:
Thanks for the suggestion. I like that!